Lowering upload threshold for data collection and adding a unit column

insha opened the pull request:
Reducing the threshold for uploading data that is collected using the data collector to 50k file size or 24  hours; whichever is reach first for it to be uploaded. [APC-2927]

Commit:
[APC-2927] Adding a column for unit.

Commit:
Reducing threshold for collected data upload.

Commit:
Using the APCDataUploadAndArchiver

This will archive and upload the files that were create by the data collectors.

jwe-apple commented on APCAppCore/APCAppCore/Library/DataArchiverAndCollector/APCPassiveDataCollector.m:
```diff
@@ -309,20 +218,31 @@ - (void) APCDataTracker:(APCDataTracker *)tracker hasNewData:(NSArray *)dataArra
+    
+    if ([[NSDate date] timeIntervalSinceDate:startDate] >= tracker.stalenessInterval)
+    {
         [self flush:tracker];
```
This looks like this can flush twice in a row. Is that benign?

insha replied:
I have updated the method to trigger flush only once, when either of the two (file size or time limit) conditions are met. The changes will be reflected on the pull request shortly.

jwe-apple commented on APCAppCore/APCAppCore/Library/DataArchiverAndCollector/APCPassiveDataCollector.m:
```diff
@@ -154,108 +151,20 @@ - (void) loadOrCreateDataFiles: (APCDataTracker*) tracker
-    if (![[NSFileManager defaultManager] removeItemAtPath:unencryptedPath error:&deleteError]) {
-        APCLogError2(deleteError);
+    if (!successfullyMoved) {
+        APCLogError2(flushError);
```
Is there any other recovery possible at this point?

Have you ever observed this to fail?

insha replied:
Haven't seen it fail yet, but when it does, the only thing is to not touch the data file. That way the collector will keep appending to it; and the upload will be tried again.

jwe-apple replied:
that's what I wanted to hear!

Commit:
Not doing anything to the file when the upload fails.

The collector will keep appending to file and will try to upload the next time around.

Commit:
Only calling the flush method once.

When either the file size limit or the staleness interval is reached.

jwe-apple commented:
With that last change I'll be happy

Commit:
CamelCasing the hasReachedStalenessInterval variable.

insha commented:
Cool. Done. You should see it in a few moments.
